src/parser.yy

```diff
@@ -167,7 +167,22 @@ probes : probes probe { $$ = $1; $1->push_back($2); }
        | probe        { $$ = new ast::ProbeList; $$->push_back($1); }
        ;
 
-probe : attach_points pred block { $$ = new ast::Probe($1, $2, $3); }
+probe : attach_points pred block { if (!driver.listing_)
+                                     $$ = new ast::Probe($1, $2, $3);
+                                   else
+                                   {
+                                     error(@$, "unexpected listing query format");
+                                     YYERROR;
+                                   }
+                                 }
+      | attach_points END { if (driver.listing_)
+                              $$ = new ast::Probe($1, nullptr, nullptr);
+                            else
+                            {
+                              error(@$, "unexpected end of file, expected {");
```

Comment:
```
% sudo ./src/bpftrace -l 'k:vfs_open // {}'
stdin:1:12-13: ERROR: syntax error, unexpected end predicate, expecting {
k:vfs_open // {}
           ~
```

In this case, "unexpected listing query format" is more appropriate?

Reply:
True, but since this is an automatic error message by bison, I haven't yet found a way to implement it in a short way. And I don't like the idea of making large changes to the parser (potentially introducing errors) just for the sake of fixing one error message.

Reply:
I see. I also don't familiar with bison, and the current behavior is OK to me.